### PR TITLE
 New "I see an entry in the something table with:" method

### PR DESCRIPTION
**Pull request by benvand**
* New "I see an entry in the something table with:" method
* Update *broken* feature to use new 'entry exists in table' method

### CODE REVIEW

File: features/step_definitions/common_steps.rb [hunk 1]
```diff
@@ -336,6 +336,24 @@
   end
 end
 
+Then /^I see an entry in the '(.*)' table with:$/ do |table_heading, table|
+  result_table_rows = get_table_rows_by_caption(table_heading)
+  expected_row = table.rows[0]
+  for row in result_table_rows
+    result_columns = row.all('td')
+    match = true
+    for value, index in expected_row.each_with_index
```
Comment (TheDoubleK) [inline features/step_definitions/common_steps.rb]: The nested loops and overwriting the "match" hurt my head a bit, but it's doing the right thing and I can't immediately suggest a neater way to do it...

Maybe a couple of comments would help, like "checking each field of a row", "reset if a row doesn't match", etc.

File: features/step_definitions/common_steps.rb [hunk 2]
```diff
@@ -336,6 +336,24 @@
...
+      if match and value != '<ANY>' and value != result_columns[index].text
```
Comment (CrystalPea) [inline features/step_definitions/common_steps.rb]: if match is overwritten to true by us do we need to check for it in the if statement?

File: features/step_definitions/common_steps.rb [hunk 3]
```diff
@@ -336,6 +336,24 @@
   end
 end
 
+Then /^I see an entry in the '(.*)' table with:$/ do |table_heading, table|
+  result_table_rows = get_table_rows_by_caption(table_heading)
+  expected_row = table.rows[0]
+  for row in result_table_rows
```
Comment (CrystalPea) [inline features/step_definitions/common_steps.rb]: You can also use 'each' loop, which is more idiomatic to Ruby, but then as we are a Python team, 'for' loop might be preferable for readability, your call really :).

https://stackoverflow.com/questions/3294509/for-vs-each-in-ruby
https://teamtreehouse.com/library/iteration-with-each

File: features/step_definitions/common_steps.rb [hunk 4]
```diff
@@ -336,6 +336,25 @@
   end
 end
 
+Then /^I see an entry in the '(.*)' table with:$/ do |table_heading, table|
+  expected_row = table.rows[0]
+  result_table_rows = get_table_rows_by_caption(table_heading)
+  match = false
+
+  for row in result_table_rows
```
Comment (CrystalPea) [inline features/step_definitions/common_steps.rb]: You can also use 'each' loop, which is more idiomatic to Ruby, but then as we are a Python team, 'for' loop might be preferable for readability, your call really :).

https://stackoverflow.com/questions/3294509/for-vs-each-in-ruby
https://teamtreehouse.com/library/iteration-with-each

**Comment by benvand** [comment]
@CrystalPea Realised what you meant this morning 😂  Fixed that

Thanks for the reviews @TheDoubleK @CrystalPea ! Merging

**Pull request merged**